File: LeetCode/9.palindrome-number.cpp
/*
 * @lc app=leetcode id=9 lang=cpp
 *
 * [9] Palindrome Number
 */

// @lc code=start
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:

    bool isPalindrome(int x) {
        if (x<0 ){
            return false;
        }
        if (x < 10){ 
            return true;
        }

        vector<int> v;

        while (x > 9){
            v.push_back(x%10);
            x/=10;
        }
        v.push_back(x);


        for (int i = 0;i< (v.size()/2);i++){
            //cout<<v[i]<<" "<<v[v.size()-i-1];
            if (v[i] != v[v.size()-i-1]){
                return false;
            }
        }
        return true;

        
    }
};
// @lc code=end

File: LeetCode/9.palindrome-number.py
# coding: utf-8
from string import *
from re import *
from datetime import *
from collections import *
from heapq import *
from bisect import *
from copy import *
from math import *
from random import *
from statistics import *
from itertools import *
from functools import *
from operator import *
from io import *
from sys import *
from json import *
from builtins import *

import string
import re
import datetime
import collections
import heapq
import bisect
import copy
import math
import random
import statistics
import itertools
import functools
import operator
import io
import sys
import json

from typing import *

#
# @lc app=leetcode id=9 lang=python3
#
# [9] Palindrome Number
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if (x < 0):
            return False
        i  = x
        invx  = 0
        while (i != 0):
            invx = invx *10 + i%10
            i//=10

        
        return invx == x


# @lc code=end
    
def main():
    s = Solution()
    i = s.isPalindrome(20)
    print (i)

if __name__ == "__main__":
    main()



    

